### PR TITLE
Replace NOT operator with explicit `false` check - part 4

### PR DESCRIPTION
**Pull request by pugnascotia**
Part 4.

We have an in-house rule to compare explicitly against `false` instead
of using the logical not operator (`!`). However, this hasn't
historically been enforced, meaning that there are many violations in
the source at present.

We now have a Checkstyle rule that can detect these cases, but before we
can turn it on, we need to fix the existing violations. This is being
done over a series of PRs, since there are a lot to fix.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java [hunk 1]
```diff
@@ -314,7 +314,7 @@ public void validateReferences(Version indexVersionCreated, Function<String, Map
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-        if (!super.equals(o)) return false;
```
Comment (breskeby) [inline server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java:317]: Do we have plans to enforce the idea templates for generating these equals methods?

**Review by breskeby — APPROVED** [review]
looks good to me, just a question out of curiosity

File: server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java (hunk 1 above)
Comment (pugnascotia) [reply server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java:317]: Not right now - do you happen to know whether this is possible?

**Comment by pugnascotia** [comment]
Backported to `7.x` in cb9382fc7cc

File: server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java (hunk 1 above)
Comment (breskeby) [reply server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java:317]: I would expect it is somehow possible to enforce via the idea-ext plugin (https://github.com/JetBrains/gradle-idea-ext-plugin/wiki) and defining a custom template. But not for sure.

Comment (pugnascotia) [reply server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java:317]: So actually IntelliJ seems to prefer using `Object.equals` when generating `equals()` methods, so I don't think we have do anything.

**Pull request merged**